lint: Add Prettier for Glint templates

backspace opened the pull request:
This adds a root Prettier config with the necessary plugin and setup to allow Prettier to run on `gts` files. That means a lot of lint-only changes, including all of `packages/realm-server`, which had no config and therefore defaulted to double quotes, where other packages were configured for single quotes.

This also fixes longstanding check annotations that were showing on all PRs, like [here](https://github.com/cardstack/boxel-motion/pull/243/files#diff-0a0188e27e9161ddf75ef6d4f21b48200d520ce49d4748b5c73efbb6ed27ad6f).

backspace committed:
Add Prettier autofixes for host Glint templates

This includes these tweaks from @wagenet:
https://github.com/gitKrystan/prettier-plugin-ember-template-tag/issues/38#issue-1462445460

backspace committed:
Add Glint Prettier autofixes for boxel-ui

backspace committed:
Add Glint Prettier autofixes for realm-server

backspace committed:
Fix one realm server test

I can’t get the other one working 🫠

backspace committed:
Add Prettier autofixes for other templates

Adding ESLint/Prettier to the root has increased the
chaos significantly, hopefully I can recover…

backspace committed:
Add filter on root lint

This prevents commands in vendor from running.

backspace committed:
Remove inapplicable note

backspace committed:
Update from main

backspace committed:
Add Prettier Glint autofixes again

ef4 commented:
If it works correctly with a single prettier config in the monorepo root, let's do that.

If not, then we can explicitly setup prettier in those workspaces.

backspace committed:
Merge branch 'main' into glint-prettier-cs-4836

# Conflicts:
#	packages/base/card-api.gts
#	packages/base/catalog-entry.gts
#	packages/base/contains-many-component.gts
#	packages/base/default-card-component.gts
#	packages/base/field-component.gts
#	packages/base/links-to-editor.gts
#	packages/base/shadow-dom.gts
#	packages/boxel-ui/addon/components/input.gts
#	packages/boxel-ui/addon/components/label.gts
#	packages/demo-cards/address.gts
#	packages/demo-cards/asset.gts
#	packages/demo-cards/invoice-packet.gts
#	packages/demo-cards/payment-method.gts
#	packages/demo-cards/person.gts
#	packages/demo-cards/pet.gts
#	packages/demo-cards/vendor.gts
#	packages/host/tests/integration/modifiers/attach-styles-test.gts

backspace committed:
Add autofixes again

It’s not working in packages/host 🤔
`No parser could be inferred for file`

backspace committed:
Update lockfile

backspace committed:
Remove nested Prettier config files

backspace committed:
Add more autofixes

backspace committed:
Merge branch 'main' into glint-prettier-cs-4836

backspace committed:
Remove merge error

This is now gone on the main branch.

backspace committed:
Update realm server files with single quotes

This was a package that didn’t have a Prettier config
so it used the default double quotes.

backspace committed:
Fix various long-standing check annotations

backspace committed:
Move ignore

This was shifted by autoformat.

backspace committed:
Update fixture

backspace committed:
Add fix for expected opcodes

This was a guessing game!

backspace committed:
Merge branch 'main' into glint-prettier-cs-4836

# Conflicts:
#	packages/realm-server/tests/realm-server-test.ts

backspace committed:
Add realm server test autofixes again

backspace committed:
Fix id again

backspace committed:
Change expected quotes
